cirq/ops/pauli_string.py

```diff
@@ -259,6 +259,20 @@ def keys(self) -> KeysView[raw_types.Qid]:
     def qubits(self) -> Tuple[raw_types.Qid, ...]:
         return tuple(sorted(self.keys()))
 
+    def _circuit_diagram_info_(self, args: 'cirq.CircuitDiagramInfoArgs'
+                              ) -> List[str]:
+        if not len(self._qubit_pauli_map):
+            return NotImplemented
+
+        qs = args.known_qubits or list(self._qubit_pauli_map.keys())
+        symbols = list(str(self.get(q)) for q in qs)
+        prefix = ('+' if self.coefficient == +1 else '-' if
+                  self.coefficient == -1 else 'i' if self.coefficient == 1j else
+                  '-i' if self.coefficient == -1j else repr(self.coefficient) +
```

Comment:
The case of PauliStrings outside the Pauli group, i.e. with coefficients not in {1, -1, i, -i} may render a long ugly floating point number. Perhaps add explicit format string, e.g. limiting to three or four decimal places.

Reply:
Done.